homeassistant/components/walkingpad/__init__.py

```diff
@@ -0,0 +1,500 @@
+"""Support for Xiaomi WalkingPad treadmill."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, Final
+
+from bleak.exc import BleakError
+from miio.exceptions import DeviceException
+from miio.walkingpad import OperationMode, Walkingpad, WalkingpadStatus
+from ph4_walkingpad.pad import Controller, WalkingPad
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_TOKEN, STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import (
+    AWAIT_SLEEP_INTERVAL,
+    CONF_CONN_TYPE,
+    CONF_DEFAULT_SPEED,
+    CONF_UUID,
+    DEFAULT_SPEED,
+    DOMAIN,
+    MAX_SPEED,
+    MIN_SPEED,
+    MODES_DICT,
+)
+
+PARALLEL_UPDATES: Final = 1
+
+PLATFORMS: Final = ["remote"]
+
+_LOGGER = logging.getLogger(__name__)
+
+# turn off ph4-walkingpad logger
+logging.getLogger("ph4_walkingpad.pad").setLevel(logging.CRITICAL)
+logging.getLogger("ph4_walkingpad").setLevel(logging.CRITICAL)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WalkingPad device from a config entry."""
+    device = (
+        WalkingPadBLEDevice(hass, entry)
+        if entry.data[CONF_CONN_TYPE] == "ble"
+        else WalkingPadWiFiDevice(hass, entry)
+    )
+
+    await device.async_setup()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = device
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    async def update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Handle options update."""
+        device._default_speed = entry.options.get(CONF_DEFAULT_SPEED, DEFAULT_SPEED)
+
+    entry.async_on_unload(entry.add_update_listener(update_listener))
+
+    return True
+
+
+class WalkingPadBaseDevice:
```

Comment:
Could you move all these classes to a separate module (eg. `models.py`)?
It would help clarify the code.